### PR TITLE
Ensure only georeferenced rasters are both "Maps" and "Datasets"

### PR DESCRIPTION
**Pull request by thatbudakguy**
Part of #1514

This changes the indexing strategy for resource class to use a
MatchMap to make it much simpler to get map/dataset values out
of the data.

It also ensures that any georeferenced images are counted as both
maps and datasets.

### CODE REVIEW

File: lib/translation_maps/geo_resource_class.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
```
Comment (marlo-longley) [inline lib/translation_maps/geo_resource_class.rb:1]: Wowww thanks MatchMap!

File: spec/integration/geo_aardvark_config_spec.rb [hunk 2]
```diff
@@ -320,7 +319,7 @@
     let(:druid) { 'bb021mm7809' }
 
     it 'maps the resource class' do
-      expect(result['gbl_resourceClass_sm']).to eq %w[Datasets Maps]
+      expect(result['gbl_resourceClass_sm']).to eq ['Datasets']
```
Comment (marlo-longley) [inline spec/integration/geo_aardvark_config_spec.rb:322]: Aren't we supposed to test that these are classed as both Maps and Datasets?

File: lib/translation_maps/geo_resource_class.rb (hunk 1 above)
Comment (thatbudakguy) [reply lib/translation_maps/geo_resource_class.rb:1]: It's perfect for this! Bill clearly has done a lot of this type of thing, including in traject.

File: spec/integration/geo_aardvark_config_spec.rb (hunk 2 above)
Comment (thatbudakguy) [reply spec/integration/geo_aardvark_config_spec.rb:322]: Nope – this one is not georeferenced, so it's only a dataset.

**Review by marlo-longley — APPROVED** [review]

**Pull request merged**